compose/project.py

```diff
@@ -412,10 +440,12 @@ def down(
         self.remove_images(remove_image_type)
 
     def remove_images(self, remove_image_type):
-        for service in self.get_services():
+        for service in self.services:
             service.remove_image(remove_image_type)
 
     def restart(self, service_names=None, **options):
+        # filter service_names by enabled profiles
+        service_names = [s.name for s in self.get_services(service_names)]
```

Comment:
`restart` should consider the enabled profiles so to not start services disabled by profile with `docker-compose restart`